overlord/fdestate/fdemgr.go

```diff
@@ -50,11 +51,20 @@ func Manager(st *state.State, runner *state.TaskRunner) *FDEManager {
 }
 
 func (m *FDEManager) Ensure() error {
-	return nil
+	m.state.Lock()
+	defer m.state.Unlock()
+	return ensureState(m.state)
 }
 
-func (m *FDEManager) resealKeyForBootChains(method device.SealingMethod, rootdir string, params *boot.ResealKeyForBootChainsParams, expectReseal bool) error {
-	return backend.ResealKeyForBootChains(method, rootdir, params, expectReseal)
+func (m *FDEManager) resealKeyForBootChains(locked bool, method device.SealingMethod, rootdir string, params *boot.ResealKeyForBootChainsParams, expectReseal bool) error {
+	doUpdate := func(role string, containerRole string, bootModes []string, models []secboot.ModelForSealing, tpmPCRProfile []byte) error {
+		if !locked {
+			m.state.Lock()
+			defer m.state.Unlock()
+		}
```

Comment:
This is not elegant. But I am not sure how to handle it correctly. We do resealing sometimes locked, sometimes not.

Reply:
do we know where is it called with the state really already unlocked?